### PR TITLE
WebHost, Core: Move item and location descriptions to `WebWorld` responsibilities.

### PR DESCRIPTION
**Pull request by ThePhar**
## What is this fixing or adding?
Moving these changes from `World` to `WebWorld` to remove responsibility of these from core to WebHost.

~~Also includes some optimizations.~~

## How was this tested?
Migrated tests and ran tests. Also ensured functionality remained the same.

## If this makes graphical changes, please attach screenshots.
image

### CODE REVIEW

File: docs/world api.md [hunk 1]
```diff
@@ -58,67 +58,105 @@ game.
...
+  `OptionDict`, `OptionList`, `OptionSet`, `FreeText`, or custom `Option`-derived classes are not supported for presets on the webhost at this time.
+
+  ```python
+  # presets.py
+  options_presets = {
+      "Example Preset": {
+          "progression_balancing": 0,
+          "difficulty": "hard",
+          "include_traps": False,
+      }
```
[the remark this answers is not in the document]
Comment (ThePhar) [reply docs/world api.md:99]: out of scope for PR.

**Comment by ThePhar** [comment]
Updated PR and fixed merge conflicts.

File: worlds/AutoWorld.py [hunk 2]
```diff
@@ -4,12 +4,11 @@
...
 import pathlib
 import random
-import re
 import sys
 import time
 from dataclasses import make_dataclass
-from typing import (Any, Callable, ClassVar, Dict, FrozenSet, List, Mapping,
-                    Optional, Set, TextIO, Tuple, TYPE_CHECKING, Type, Union)
+from typing import Any, Callable, ClassVar, Dict, FrozenSet, List, Mapping, Optional, Set, TextIO, Tuple, \
+                   TYPE_CHECKING, Type, Union
```
Comment (beauxq) [inline worlds/AutoWorld.py:11]: PEP8 says this:
> The preferred way of wrapping long lines is by using Python’s implied line continuation inside parentheses, brackets and braces. Long lines can be broken over multiple lines by wrapping expressions in parentheses. These should be used in preference to using a backslash for line continuation.

Comment (beauxq) [reply worlds/AutoWorld.py:11]: (The parentheses way is also my personal preference.)

Comment (ThePhar) [reply worlds/AutoWorld.py:11]: Fair enough, my personal preference shouldn't override PEP8 I suppose for something silly like this.

File: worlds/AutoWorld.py [hunk 3]
```diff
@@ -4,12 +4,11 @@
...
 import pathlib
 import random
-import re
 import sys
 import time
 from dataclasses import make_dataclass
-from typing import (Any, Callable, ClassVar, Dict, FrozenSet, List, Mapping,
-                    Optional, Set, TextIO, Tuple, TYPE_CHECKING, Type, Union)
+from typing import (Any, Callable, ClassVar, Dict, FrozenSet, List, Mapping, Optional, Set, TextIO, Tuple,
+                   TYPE_CHECKING, Type, Union)
```
Comment (beauxq) [inline worlds/AutoWorld.py:11]:
```suggestion
                    TYPE_CHECKING, Type, Union)
```

**Pull request merged**